node_modules/oae-principals/lib/definitiveDeletion.js

```diff
@@ -0,0 +1,1851 @@
+                            listElementByMember.users[j].contentWillBeDeleted.push(data[i].resource);
+                        } else if (data[i].action === null && (listElementByMember.users[j].userJustLeaving.indexOf(data[i].resource) === -1)) {
+                            listElementByMember.users[j].userJustLeaving.push(data[i].resource);
+                        }
+                    }
+                    // If user do not exist in the list, create it
+                    else if ((listElementByMember.users.length === j+1) && (exist === false)) {
+                        listElementByMember.users.push({'userWillBePromoted': [], 'contentWillBeDeleted': [], 'userJustLeaving': [], 'profile': data[i].memberList[k].profile});
+                        if (data[i].action === true) {
+                            listElementByMember.users[j+1].userWillBePromoted.push(data[i].resource);
+                        } else if (data[i].action === false) {
+                            listElementByMember.users[j+1].contentWillBeDeleted.push(data[i].resource);
+                        } else if (data[i].action === null) {
+                            listElementByMember.users[j+1].userJustLeaving.push(data[i].resource);
+                        }
+                    }
+                }
+            }
+        }
+    }
+};
+
+/**
+ * Send an email token to a user that have a resource shared with the deleted user
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {Object}     data            User to send mail + resource
+ * @param  {String}     [token]         The token to send. If left null, a new one will be generated
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var _sendEmail = function(ctx, data, callback) {
+
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'user': user.id}, 'Unable to send a user a verification email');
+        }
+    };
+
+    if (_.isEmpty(data)) {
+        return callback(null);
+    }
+
+    var userDeleted = {};
+    userDeleted = data[0].userDeleted;
+
+    // Grab the configuration field. This will return the number of months 
+    var month = PrincipalsConfig.getValue(ctx.tenant().alias, 'user', 'delete');
+        
+    // ParseData from variable informations to make an union on user
+    _parseData(data, userDeleted.id, function(err, parseData) {
+
+        if (err) {
+            return callback(err);
+        }
+
+        if (_.isEmpty(parseData)) {
+            return callback(null);
+        }
+
+        var listEmail = [];
+
+        // Send an unique email to each user
+        parseData.users.forEach(function(user, i, arrI) {
+
+            if (!(_.isEmpty(user.userWillBePromoted) && _.isEmpty(user.contentWillBeDeleted) && _.isEmpty(user.userJustLeaving))) {
```

Comment:
Here I check if one of them are not empty, if the three of them are empty I don't enter in the if condition.

Reply:
But now the condition is only entered if `user.userWillBePromoted` is not empty **AND** `user.contentWillBeDeleted` and `user.userJustLeaving` *are* empty. Is that what you meant to do? It sounds like it'd make more sense to use `||` (or) instead of `&&` (and) if I'm understanding this right?